ray-operator/controllers/ray/common/pod.go

```diff
@@ -271,7 +256,7 @@ func initLivenessAndReadinessProbe(rayContainer *corev1.Container, rayNodeType r
 			SuccessThreshold:    utils.DefaultLivenessProbeSuccessThreshold,
 			FailureThreshold:    utils.DefaultLivenessProbeFailureThreshold,
 		}
-		rayContainer.LivenessProbe.Exec = &corev1.ExecAction{Command: []string{"bash", "-c", strings.Join(commands, " && ")}}
+		rayContainer.LivenessProbe.HTTPGet = &corev1.HTTPGetAction{Path: healthCheckPath, Port: healthCheckPort}
```

Comment:
Using HTTP probes means we can only query 1 endpoint per probe now. For head pod this would `/api/gcs_healthz` and for worker pod it would be `api/local_raylet_healthz`. I'm not sure if not health checking `api/local_raylet_healthz` in the head pod is problematic, it would depend on what whether `/api/gcs_healthz` incorporates raylet health in some way as well